### ISSUE TITLE
Patch operation path

### ISSUE THREAD

**Issue by lstroud** [json-api/json-api#225]
Currently, the examples use a path of '/photos/0/links/author'. Shouldn't it be the photo's id in the path and not it's index in the array?

**Comment by dgeb** [comment]
This will be updated soon so that the operation's path is complementary (i.e. relative) to the request URL. And yes, the photo's id will be used in that example.

**Comment by gr0uch** [comment]
@dgeb is this compatible with the [JSON Patch RFC](https://tools.ietf.org/html/rfc6902)?

**Comment by guillec** [comment]
#176 #193 These issues might help you understand what is being proposed.

**Comment by dgeb** [comment]
@daliwali I believe my suggested approach is compliant with 6902, although the spec seems somewhat open to interpretation:

> Additionally, operation objects MUST have exactly one "path" member.
>    That member's value is a string containing a JSON-Pointer value
>    [RFC6901] that references a location within the target document (the
>    "target location") where the operation is performed.

The question comes down to what is considered to be the target document of a JSON Patch operation. The current examples in JSON API consider the target document to be the response document from a GET request to the same path, in which arrays are accessed by position instead of as a set keyed by id.

However, I find it much more appropriate (and consistent with other requests) to operate on the target document discussed in #193.

The following example from 6902 is consistent with this approach (or at least, doesn't contradict it):

```
   PATCH /my/data HTTP/1.1
   Host: example.org
   Content-Length: 326
   Content-Type: application/json-patch+json
   If-Match: "abc123"

   [
     { "op": "test", "path": "/a/b/c", "value": "foo" },
     { "op": "remove", "path": "/a/b/c" },
     { "op": "add", "path": "/a/b/c", "value": [ "foo", "bar" ] },
     { "op": "replace", "path": "/a/b/c", "value": 42 },
     { "op": "move", "from": "/a/b/c", "path": "/a/b/d" },
     { "op": "copy", "from": "/a/b/d", "path": "/a/b/e" }
   ]
```

The PATCH is issued against `/my/data` and the paths for each operation in the request document are clearly relative to the request path, in that they don't repeat it.

**Comment by pixelhandler** [comment]
If the path is relative to the resource it does seem unnecessary to include a reference to the resource in the path. I was confused when I saw a request for `/resource/A` and in the path `/resource/B/property`, just using `/property` or `/` in the path would be easier to understand.

**Comment by dgeb** [comment]
Closed via #237 and implemented as described above:

> This will be updated soon so that the operation's path is complementary (i.e. relative) to the request URL.
